contracts/adapters/OptimisticAuctionRebalanceExtension.sol

```diff
@@ -0,0 +1,387 @@
+        require(productSettings.rulesHash != bytes32(""), "Rules not set");
+        require(address(productSettings.optimisticParams.optimisticOracleV3) != address(0), "Oracle not set");
+
+
+        bytes memory claim = _constructClaim(proposalHash, productSettings.rulesHash);
+        uint256 totalBond = _pullBond(productSettings.optimisticParams);
+
+        bytes32 assertionId = productSettings.optimisticParams.optimisticOracleV3.assertTruth(
+            claim,
+            msg.sender,
+            address(this),
+            address(0),
+            productSettings.optimisticParams.liveness,
+            productSettings.optimisticParams.collateral,
+            totalBond,
+            productSettings.optimisticParams.identifier,
+            bytes32(0)
+        );
+
+        assertionIds[proposalHash] = assertionId;
+        proposedProduct[assertionId] = Proposal({
+            proposalHash: proposalHash,
+            product: setToken
+        });
+
+        emit RebalanceProposed( setToken, _quoteAsset, _oldComponents, _newComponents, _newComponentsAuctionParams, _oldComponentsAuctionParams, _shouldLockSetToken, _rebalanceDuration, _positionMultiplier);
+        emit AssertedClaim(setToken, msg.sender, productSettings.rulesHash, assertionId, claim);
+
+    }
+   
+    /**
+     * @dev OPERATOR ONLY: Checks that the old components array matches the current components array and then invokes the 
+     * AuctionRebalanceModuleV1 startRebalance function.
+     * 
+     * Refer to AuctionRebalanceModuleV1 for function specific restrictions.
+     *
+     * @param _quoteAsset                   ERC20 token used as the quote asset in auctions.
+     * @param _oldComponents                Addresses of existing components in the SetToken.
+     * @param _newComponents                Addresses of new components to be added.
+     * @param _newComponentsAuctionParams   AuctionExecutionParams for new components, indexed corresponding to _newComponents.
+     * @param _oldComponentsAuctionParams   AuctionExecutionParams for existing components, indexed corresponding to
+     *                                      the current component positions. Set to 0 for components being removed.
+     * @param _shouldLockSetToken           Indicates if the rebalance should lock the SetToken.
+     * @param _rebalanceDuration            Duration of the rebalance in seconds.
+     * @param _positionMultiplier           Position multiplier at the time target units were calculated.
+     */
+    function startRebalance(
+        IERC20 _quoteAsset,
+        address[] memory _oldComponents,
+        address[] memory _newComponents,
+        AuctionExecutionParams[] memory _newComponentsAuctionParams,
+        AuctionExecutionParams[] memory _oldComponentsAuctionParams,
+        bool _shouldLockSetToken,
+        uint256 _rebalanceDuration,
+        uint256 _positionMultiplier
+    )
+        external
+        override
```

Comment:
Do you need to have `onlyOperator` here?

Reply:
Not 100% sure. Ideally the UMA integration should allow us to have this open for anyone to call. 
However maybe we want to keep the training wheels on for now and make it onlyOperator ? (or `onlyAllowedCallers` which would allow us to set this to be callable by anyone later).

@pblivin0x What do you think ?

Reply:
Either way we should make it consistent with the docstring comment though of course, which currently states this to be "operator only".

Reply:
This being callable by anyone seems to have been on on purpose though with the global extensions as shown by [this test case](https://github.com/IndexCoop/index-coop-smart-contracts/blob/d5ab03004853f068528a1962d9d0b025dd8ef0e5/test/global-extensions/globalOptimisticAuctionRebalanceExtension.spec.ts#L476).

Reply:
this is fine to be public, we can update docstrings to reflect this